tests/reference_images.rs

```diff
@@ -95,49 +97,135 @@ fn check_references() {
             let mut path: Vec<_> = path.components().collect();
             (path.pop().unwrap(), path.pop().unwrap())
         };
+
+        let mut filename_parts = filename
+            .as_os_str()
+            .to_str()
+            .unwrap()
+            .split('.')
+            .collect::<Vec<_>>();
+
+        // Ignore the file extension
+        filename_parts.pop().unwrap();
+
+        // The penultimate part of `filename_parts` represents the metadata.
+        let meta_str = filename_parts.pop().unwrap();
+        let mut meta = meta_str.split('_').collect::<Vec<_>>();
+        let ref_crc;
+        let anim_frame: Option<usize>;
```

Comment:
Move these in front of the block which assigns them, without spacing. I'm not sure the type ascription is necessary but it really doesn't hurt. Together with the enum, I'd expect something like:

```
let ref_crc;
let anim_frame: Option<usize>;
match meta {
    TestKind::Crc => {
        ref_crc = meta.pop().unwrap();
        anim_frame = None;
    }
…
}
```